modules/ai/assets/js/editor/components/prompt-history/history-modal.js

```diff
@@ -0,0 +1,161 @@
+import PromptHistoryUpgrade from './parts/modal-upgrade';
+import { HISTORY_TYPES } from './history-types';
+
+const StyledContent = styled( Box )`
+	width: 360px;
+	position: relative;
+	margin-top: ${ ( { theme } ) => theme.spacing( 5 ) };
+	margin-right: ${ ( { theme } ) => theme.spacing( 5 ) };
+	background-color: ${ ( { theme } ) => theme.palette.background.paper };
+	border-radius: ${ ( { theme } ) => theme.border.radius.sm };
+	height: ${ ( { fullHeight } ) => fullHeight ? '86vh' : '52vh' };
+
+	@media screen and (max-width: 456px) {
+		width: 320px;
+	}
+
+  @media screen and (max-width: 420px) {
+		width: 230px;
+  }
+`;
+
+const ITEMS_LIMIT = 10;
+const NO_HISTORY_ACCESS_ERROR = 'invalid_connect_data';
+
+const PromptHistoryModal = ( { ...props } ) => {
+	const lastRun = useRef( () => {} );
+	const scrollContainer = useRef( null );
+	const { onModalClose, promptType } = useContext( PromptHistoryContext );
+
+	const {
+		items,
+		meta,
+		isLoading: isHistoryFetchingInProgress,
+		error: historyFetchingError,
+		send: fetchData,
+		deleteItemById,
+	} = usePromptHistory( promptType );
+
+	const {
+		isLoading: isDeletingInProgress,
+		error: historyDeletingError,
+		send: deleteItem,
+	} = useDeletePromptHistoryItem();
+
+	const error = historyFetchingError || historyDeletingError;
+	const isLoading = isHistoryFetchingInProgress || isDeletingInProgress;
+	const isLastPage = meta && meta?.currentPage === meta?.totalPages;
+
+	useEffect( () => {
+		lastRun.current = async () => fetchData( { page: 1, limit: ITEMS_LIMIT } );
+
+		lastRun.current();
+	}, [] );
+
+	const loadNext = () => {
+		if ( isLoading || isLastPage ) {
+			return;
+		}
+
+		lastRun.current = async () => fetchData( { page: meta.currentPage + 1, limit: ITEMS_LIMIT } );
+
+		lastRun.current();
+	};
+
+	const onHistoryItemDelete = async ( id ) => {
+		lastRun.current = async () => await deleteItem( id );
+
+		await lastRun.current();
+
+		if ( ! historyDeletingError ) {
+			deleteItemById( id );
+		}
+	};
+
+	const renderPeriods = () => {
+		const groupData = groupPromptHistoryData( items );
+		const periods = [];
+
+		if ( groupData[ LAST_7_DAYS_KEY ]?.items?.length ) {
+			periods.push( <PromptHistoryPeriod periodTitle={ groupData[ LAST_7_DAYS_KEY ].label }
+				onHistoryItemDelete={ onHistoryItemDelete }
+				historyItems={ groupData[ LAST_7_DAYS_KEY ].items } /> );
+		}
+
+		if ( groupData[ LAST_30_DAYS_KEY ]?.items?.length ) {
+			periods.push( <PromptHistoryPeriod periodTitle={ groupData[ LAST_30_DAYS_KEY ].label }
+				onHistoryItemDelete={ onHistoryItemDelete }
+				historyItems={ groupData[ LAST_30_DAYS_KEY ].items } /> );
+		}
+
+		for ( let i = 11; i >= 0; i-- ) {
+			if ( groupData[ i ] ) {
+				periods.push( <PromptHistoryPeriod periodTitle={ groupData[ i ].label }
+					onHistoryItemDelete={ onHistoryItemDelete }
+					historyItems={ groupData[ i ].items } /> );
+			}
+		}
+
+		return periods;
+	};
+
+	return (
+		<Modal
```

Comment:
> Why not use the Dialog component?

Semantically it doesn't seem right:

> Dialogs inform users about a task and **can contain critical information, require decisions, or involve multiple tasks**.

> A Dialog is a type of [modal](https://mui.com/material-ui/react-modal/) window that appears in front of app content to **provide critical information or ask for a decision**.

[Source](https://mui.com/material-ui/react-dialog/)


> if you must use the Modal at least extract it into a separate component

I considered this modal as separate as I don't think this particular setup might be applicable to any other modal. Wdyt?

Reply:
Basically, Dialog implements Modal and the Backdrop etc.. so you basically just created a composition of a Dialog and the result is probably very close anyway.., so I think that the consideration should be related to: if the Dialog component already handles some of the behaviors for you and saving you code instead of creating the composition yourself.